### ISSUE TITLE
Nested entity map hydration allow for custom closure hydrator

### ISSUE THREAD

**Issue by zakhenry** [spira/spira#247]
@jeremysik I've been thinking on the best way to handle the custom requirement that we have for the nested article metas to use the template for hydrating the collection, and I think the best way is to extend the `nestedEntityMap` to have the following interface:

```
interface INestedEntityMap {
    [key:string] : IModelClass | ():any;
}
```

Then we can extend the `hydrateNested` method to inspect the type of the nested entity value (this is actually a good use case for the new typescript 1.6 type check function) and either hydrate directly with the model class, or just use the closure.

```
public static isModelClass(entity: any): entity is IModelClass {
  return entity.prototype && entity.prototype instanceOf AbstractModel;
}
```

**Comment by ghost** [comment]
I will implement this as discussed @xiphiaz.

**Comment by zakhenry** [comment]
Closed by #249
